filesystem/Folder.php

```diff
@@ -258,11 +258,24 @@ public function getAbsoluteURL() {
 	public function onAfterWrite() {
 		parent::onAfterWrite();
 
-		// Ensure that children loading $this->Parent() load the refreshed record
-		$this->flushCache();
+		// No publishing UX for folders, so just cascade changes live
+		if(Versioned::get_stage() === Versioned::DRAFT_STAGE) {
+			$this->publish(Versioned::DRAFT_STAGE, Versioned::LIVE_STAGE);
+		}
+
+		// Update draft version of all child records
 		$this->updateChildFilesystem();
 	}
 
+	public function onAfterDelete() {
+		parent::onAfterDelete();
+
+		// Cascade deletions to live
+		if(Versioned::get_stage() === Versioned::DRAFT_STAGE) {
+			$this->deleteFromStage(Versioned::LIVE_STAGE);
```

Comment:
Yeah, the problem is Folder extends File, but we want versioning on Files, and not necessarily Folders...